tools/osbuild-mpp

```diff
@@ -158,6 +158,40 @@ of the image, otherwise the "latest" tag is used. If "name" is specified
 that is used as the custom name for the container when installed.
 
 
+OSTree commit resolving:
+
+This tool consumes the `mpp-resolve-ostree-commits` option and produces
+source and inputs entries for the ostree commits.
+
+It supports version version "2" of the manifest description format.
+
+The parameters for this pre-processor, version "2", look like this:
+
+```
+...
+    {
+      "name": "org.osbuild.deploy",
+      ...
+      "inputs": {
+        "commits": {
+          "mpp-resolve-ostree-commits": {
+            "commits": [
+              {
+                 "ref": "fedora/x86_64/coreos/stable"
+                 "remote": {
+                      "url": https://kojipkgs.fedoraproject.org/ostree/repo/
```

Comment:
This always uses the same ref in both the generated source and the operation (like deploy). Might you not sometimes want to use a different name? For example, imagine you want the generated image to follow `fedora/x86_64/coreos/stable` as an update path, but you want the manifest to specifically contain the content of a particular version like `fedora/x86_64/coreos/v17.42`

Reply:
I'm fuzzy here, but is the remotes file actually written during the deploy? I thought it came from a file inside the commit (and sometimes overwritten).

Reply:
> This always uses the same ref in both the generated source and the operation (like deploy). Might you not sometimes want to use a different name?

Maybe? Today, for sources you specify a commit and a remote:

```
"org.osbuild.ostree": {
  "items": {
    "456b999877532e83e40b5a292def550447d76b1c91cb4340ff869f1b7516d404": {
      "remote": {
        "url": "https://kojipkgs.fedoraproject.org/ostree/repo/"
      }
    }
  }
}

```

In that case there is no ref that is passed around, you'd set it in the `ostree.pull`. What I'm mostly trying to do here is make it so I don't have to go do the "rev-parse" operation myself when creating a manifest. I know there are higher level tools for this, but since mpp-resolve-images existed I figured an analogue here for ostree made sense.

Maybe we add it as an enhancement in the future if we determine there is demand for it?

> I'm fuzzy here, but is the remotes file actually written during the deploy? I thought it came from a file inside the commit (and sometimes overwritten).

I think you are talking about the origin file:


```
[core@rpi4 ~]$ cat /sysroot/ostree/deploy/fedora-coreos/deploy/14a36bafa9d3bee7c3ea8f17e3fdb45a76781e76eff53597e1e4497f2cf7b25b.0.origin
[origin]
baserefspec=fedora:fedora/aarch64/coreos/next
override-commit=6203cbeb698f6a726165239374ef80c15d7c7fd0057bc582cf03501218074765
```

OSBuild doesn't currently do anything with that right now, but the values in there do get inherited from the `ostree admin deploy`.

Reply:
> Maybe we add it as an enhancement in the future if we determine there is demand for it?

That could work too. Although the mpp-resolve-image feature does allow this kind of "rename" already. I.e. if you just set source as docker.io/library/ubuntu, then that will be the local name of the image, but you can also set name to give it a different name in the local store. So, there is some precedence for this.

That one is similar in what it expands too also. 

Here is an example:
```
- type: org.osbuild.skopeo
  inputs:
    images:
      type: org.osbuild.containers
      origin: org.osbuild.source
      mpp-resolve-images:
        images:
          - source: registry.gitlab.com/centos/automotive/sample-images/demo/auto-apps
            tag: latest
            name: localhost/auto-apps
```

It expands to a source:
```
"org.osbuild.skopeo": {
  "items": {
    "sha256:6667bb22d02f91dc30a81841a1ce46c8dced20f2f300afb76921f48602efddb7": {
      "image": {
        "name": "registry.gitlab.com/centos/automotive/sample-images/demo/auto-apps",
        "digest": "sha256:285beba3acccdda8912ee3f5598df655364c2ac8e0c74c5b4d77caef49f5f12c"
      }
    }
  }
}
```

And then a stage using the source as input:
```
"inputs": {
  "images": {
    "type": "org.osbuild.containers",
    "origin": "org.osbuild.source",
    "references": {
      "sha256:6667bb22d02f91dc30a81841a1ce46c8dced20f2f300afb76921f48602efddb7": {
        "name": "localhost/auto-apps"
      }
    }
  }
}
```

In this example we're renaming the image from the full registry name to an un-prefixed version.

Reply:
That's fair. I'll try to implement something like this (maybe not the exact same) and do a new upload.

Reply:
Seems like this is covered then.  Should we resolve this thread?

Reply:
I'll mark it resolved. @alexlarsson chime in if it's not sufficient.